### PR TITLE
Add API docs swagger json

### PR DESCRIPTION
**Pull request by karniv00l**
`/api/doc`

Screenshot 2020-09-10 at 11 47 08

### CODE REVIEW

File: tests/Functional/Controller/Api/PackageControllerTest.php [hunk 1]
```diff
@@ -40,11 +40,12 @@ public function testAuthorizationRequired(): void
             $this->lastResponseBody(),
             '
             {
-                "errors": {
-                    "credentials": [
-                        "Authentication required."
-                    ]
-                }
```
Comment (akondas) [inline tests/Functional/Controller/Api/PackageControllerTest.php:47]: :thinking: prev format looks better ... is it forced by something changing?

File: composer.json [hunk 2]
```diff
@@ -48,7 +52,10 @@
         "symfony/messenger": "*",
         "symfony/monolog-bundle": "^3.5",
         "symfony/process": "*",
+        "symfony/property-access": "5.1.*",
+        "symfony/property-info": "5.1.*",
         "symfony/security-bundle": "*",
+        "symfony/serializer": "5.1.*",
```
Comment (akondas) [inline composer.json:58]: please use `*`, this will allow to update all symfony components together

File: config/packages/nelmio_api_doc.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+nelmio_api_doc:
+    models: { use_jms: false }
+
+    documentation:
+        info:
+            title: Private PHP Package Repository Manager
+            description: ...
```
Comment (akondas) [inline config/packages/nelmio_api_doc.yaml:7]:
```
title: Repman API
description: Private PHP Package Repository Manager
```

File: config/routes/nelmio_api_doc.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Expose your documentation as JSON swagger compliant
+app.swagger:
+    path: /api/doc.json
+    methods: GET
+    defaults: { _controller: nelmio_api_doc.controller.swagger }
```
Comment (akondas) [inline config/routes/nelmio_api_doc.yaml:5]: what about UI?
```
app.swagger_ui:
    path: /api/doc
    methods: GET
    defaults: { _controller: nelmio_api_doc.controller.swagger_ui }
```
or maybe something else?

**Comment by akondas** [comment]
Tested with Swagger UI, looks good :+1: 
Screenshot from 2020-09-10 08-43-45

File: config/routes/nelmio_api_doc.yaml (hunk 4 above)
Comment (karniv00l) [reply config/routes/nelmio_api_doc.yaml:4]: added

File: tests/Functional/Controller/Api/PackageControllerTest.php (hunk 1 above)
Comment (karniv00l) [reply tests/Functional/Controller/Api/PackageControllerTest.php:47]: It's a problem to describe schema when having dynamic keys

File: composer.json [hunk 5]
```diff
@@ -121,6 +123,9 @@
         }
     ],
     "scripts": {
+        "post-install-cmd": [
+            "bin/console assets:install --symlink"
+        ],
```
Comment (karniv00l) [inline composer.json:128]: https://symfony.com/doc/master/bundles/NelmioApiDocBundle/faq.html#asset-files-not-loaded

File: tests/Functional/Controller/Api/PackageControllerTest.php (hunk 1 above)
Comment (akondas) [reply tests/Functional/Controller/Api/PackageControllerTest.php:47]: good argument :+1:

**Review by akondas — APPROVED** [review]

**Pull request merged**